Quick-Start [DA]/plpgsql.md

```diff
@@ -0,0 +1,271 @@
+4. **Performance Optimization:** By choosing the right procedural language for specific tasks, developers can achieve performance improvements over SQL, especially for complex data operations.
+
+To use procedural languages in PostgreSQL:
+
+1. **Enable the Language:** Before using a procedural language in a database, it needs to be enabled using the `CREATE LANGUAGE` command or by installing the relevant extensions.
+
+2. **Write Functions and Procedures:** After enabling the language, developers can write functions and stored procedures using the chosen procedural language. This allows them to implement complex business logic, data transformations, and data analysis directly within the database.
+
+## Quick Start: Creating a Postgres Extension using PL/pgSQL
+
+In this quick start guide, we will create a simple PostgreSQL extension using [PL/pgSQL](https://www.postgresql.org/docs/current/plpgsql.html). The extension will include a PL/pgSQL function to subtract two numbers.
+
+### Prerequisites
+
+- [PostgreSQL](https://www.postgresql.org/download/) installed and running on your system.
+- Basic knowledge of SQL and PL/pgSQL programming.
+
+### Step 1: Create the Extension Directory
+
+Create a new directory for your extension project. You can choose any location on your system. Let's assume you create the directory in your home folder:
+
+```bash
+mkdir ~/pg_plpgsql_ext
+cd ~/pg_plpgsql_ext
+```
+
+### Step 2: Create the SQL Script File:
+
+Next, we'll create the SQL script file that defines the extension's functions and features. Create a file named `pg_plpgsql_ext--1.0.sql` in the `pg_plpgsql_ext` directory with the following content:
+
+```sql
+-- pg_plpgsql_ext--1.0.sql
+-- Create the function that uses PL/pgSQL
+CREATE OR REPLACE FUNCTION subtract_numbers(a integer, b integer)
+RETURNS integer
+LANGUAGE plpgsql
+AS $$
+BEGIN
+    RETURN a - b;
+END;
+$$;
+```
+
+#### Explanation of the SQL Script:
+- `CREATE OR REPLACE FUNCTION subtract_numbers(a integer, b integer) RETURNS integer LANGUAGE plpgsql AS $$`: This statement creates the function `subtract_numbers` with two integer arguments (`a` and `b`). The function will return an integer. The `LANGUAGE plpgsql` specifies that the function is written in PL/pgSQL.
+- `BEGIN`: This is the beginning of the PL/pgSQL code block.
+- `RETURN a - b;`: This line calculates the subtraction of `b` from `a` and returns the result.
+
+### Step 3: Create the Control File:
+
+Create a file named "pg_plpgsql_ext.control" in the "pg_plpgsql_ext" directory with the following content:
+
+```control
+# pg_plpgsql_ext.control
+
+# Extension name
+comment = 'A simple PostgreSQL extension using PL/pgSQL.'
+
+# Default version of the extension
+default_version = '1.0'
+
+# Extension is relocatable
+relocatable = true
+```
+- For more detailed information on Control Files, you can refer to the [PostgreSQL documentation on Control Files](https://www.postgresql.org/docs/current/extend-extensions.html#id-1.8.3.20.11).
```

Comment:
You can also like to your other tutorial's section that introduces the control file.